src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceApiServiceSpec.scala

```diff
@@ -1031,3 +968,93 @@ class WorkspaceApiServiceSpec extends BaseServiceSpec with WorkspaceApiService w
     }
   }
 }
+
+/**
+  * This class represents a version of the application with Workspace APIs that can run a single request.
+  */
+class MockWorkspaceApiServiceApp(val request: HttpRequest) extends MockApplication with WorkspaceApiService {
+  val workspaceServiceConstructor: (WithAccessToken) => WorkspaceService = WorkspaceService.constructor(app)
+  val permissionReportServiceConstructor: (UserInfo) => PermissionReportService = PermissionReportService.constructor(app)
+  def checkRequest: (HttpResponse, StatusCode) = {
+    request ~> dummyUserIdHeaders("1234") ~> sealRoute(workspaceRoutes) ~> check {(response, status)}
+  }
+}
+
+/**
+  * Tests that require internal state checking need to use this pattern.
+  */
+class StatefulWorkspaceApiServiceSpec extends FreeSpec with ScalaFutures with ScalatestRouteTest with Matchers with BeforeAndAfterEach {
+
+  private final val workspacesRoot = FireCloudConfig.Rawls.authPrefix + FireCloudConfig.Rawls.workspacesPath
+  private final val updateAttributesPath = workspacesRoot + "/%s/%s/updateAttributes".format(workspace.namespace, workspace.name)
+  private final val setAttributesPath = workspacesRoot + "/%s/%s/setAttributes".format(workspace.namespace, workspace.name)
+
+  "Workspace updateAttributes tests" - {
+
+    "when calling PATCH on workspaces/*/*/updateAttributes path" - {
+
+      "should 200 OK if the payload is ok" in {
+        val request = Patch(updateAttributesPath,
+          HttpEntity(MediaTypes.`application/json`, """[
+                                                      |  {
+                                                      |    "op": "AddUpdateAttribute",
+                                                      |    "attributeName": "library:dataCategory",
+                                                      |    "addUpdateAttribute": "test-attribute-value"
+                                                      |  }
+                                                      |]""".stripMargin))
+        val app = new MockWorkspaceApiServiceApp(request)
+        val (response, status) = app.checkRequest
```

Comment:
Highly optional:

You can eliminate the unused val annotation by doing
```
val (_, status) = app.checkRequest
```
(Or make a `checkOK`.)

Reply:
Yes, I was thinking about different things to return here and kept the two implementations the same for no really good reason. Some tests want to know about the response, some might need the status, and others still might need just the entity, even though I didn't add that in here. I'm leaning toward leaving the details of `checkRequest` up to the individual mock.